{gama-tui,gocovsh,golds,pkgsite,tcount}: init

phanirithvij opened the pull request:
- go
  - gama-tui  (gama was taken)
  - golds
  - pkgsite
  - gocovsh
- rust
  - tcount

| package       | version                    | source                                        | reason                                                                              |
| ----------------- | -------------------------- | ----------------------------------------- | ------------------------------------------------------------------------- |
| gama-tui      | 1.1.4  | https://github.com/termkit/gama               | github actions tui |
| gocovsh       | 0.6.1                                        | https://github.com/orlangure/gocovsh      | go code coverage tui |
| golds            | 0.7.1                                        | https://github.com/go101/golds                  | godoc/pkgsite alternative |
| pkgsite         | 0-unstable-2024-12-06        | https://github.com/golang/pkgsite             | official pkg.go.dev tool for rendering docs (godoc successor) |
| tcount          | 0-unstable-2023-04-20        | https://github.com/rrethy/tcount                 | code loc counter like tokei/scc |

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

phanirithvij committed:
tcount: init at 0-unstable-2023-04-20

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
Co-authored-by: Kenichi Kamiya <kachick1@gmail.com>

phanirithvij committed:
pkgsite: init at 0-unstable-2024-12-06

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
Co-authored-by: Kenichi Kamiya <kachick1@gmail.com>

phanirithvij committed:
golds: init at 0.7.1

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
Co-authored-by: Kenichi Kamiya <kachick1@gmail.com>

phanirithvij committed:
gocovsh: init at 0.6.1

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
Co-authored-by: Kenichi Kamiya <kachick1@gmail.com>

phanirithvij committed:
gama-tui: init at 1.1.4

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
Co-authored-by: Kenichi Kamiya <kachick1@gmail.com>

phanirithvij commented:
@ofborg build gama-tui gocovsh golds pkgsite tcount

phanirithvij commented:
I hope you don't mind reviewing.

from https://github.com/NixOS/nixpkgs/pull/358037

kachick commented on pkgs/by-name/ga/gama-tui/package.nix:
```diff
@@ -0,0 +1,40 @@
+    description = "Manage your GitHub Actions from Terminal with great UI";
+    homepage = "https://github.com/termkit/gama";
+    changelog = "https://github.com/termkit/gama/releases";
+    license = lib.licenses.gpl3Only;
```
```suggestion
    license = lib.licenses.gpl3Plus;
```

?

https://github.com/termkit/gama/blob/5063d35e48f3118fa793965fcbb931ef3b330f02/README.md?plain=1#L99

phanirithvij replied:
Used [nix-init](https://github.com/nix-community/nix-init) thanks for catching it

kachick commented on pkgs/by-name/go/gocovsh/package.nix:
```diff
@@ -0,0 +1,38 @@
+    description = "Go Coverage in your terminal: a tool for exploring Go Coverage reports from the command line";
+    homepage = "https://github.com/orlangure/gocovsh";
+    changelog = "https://github.com/orlangure/gocovsh/releases";
+    license = lib.licenses.gpl3Only;
```
:heavy_check_mark: They clarified `only` in release definition https://github.com/orlangure/gocovsh/blob/8880bc63283c13a1d630ce3817c7165a6c210d46/.goreleaser.yaml#L33

kachick commented on pkgs/by-name/tc/tcount/package.nix:
```diff
@@ -0,0 +1,33 @@
+    description = "Count your code by tokens and patterns in the syntax tree. A tokei/scc/cloc alternative";
+    homepage = "https://github.com/rrethy/tcount";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.unix;
```
https://github.com/RRethy/tcount/blob/341d9aa29512257bf7dfd7e843d02fdcfd583387/README.md?plain=1#L43

>Mac or Linux

They describes only supports Linux + MacOS
So excluding other Unix family might be stable?

```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

https://github.com/NixOS/nixpkgs/blob/960c72853a13924b2f966345ceb5c0951c4f0d55/lib/systems/doubles.nix#L111
https://github.com/NixOS/nixpkgs/blob/8f77bbf229564cdad33748822219e0fb1474e144/lib/systems/inspect.nix#L94

phanirithvij commented:
will rebase

phanirithvij commented:
```console
$ nix build -f default.nix gama-tui tcount gocovsh golds pkgsite --print-out-paths
/nix/store/xmx7bg5rlpfgd24bn9z9g2052plhyh3p-gama-tui-1.1.4-unstable-2024-09-05
/nix/store/wma3fhfnynykr5ynwhq2h5jcpkh275bd-tcount-0-unstable-2023-04-20
/nix/store/wr7xwn521xqc99shc7mjgzshjpjs3818-gocovsh-0.6.1
/nix/store/v6l3qa58lmam0nvng32xp7a1rnn5382f-golds-0.7.1
/nix/store/injqnlfzczjrzx0lap3s29v1rggk1jhs-pkgsite-0-unstable-2024-11-13
```

kachick commented on pkgs/by-name/go/golds/info_module-gover.patch:
```diff
@@ -0,0 +1,25 @@
+ 		versionData, err = ioutil.ReadFile(filepath.Join(goroot, "VERSION.cache"))
+ 	}
+ 	if err != nil {
+-		return "", fmt.Errorf("failed to get Go toolchain version in GOROOT (%s)", goroot)
++		cmd := exec.Command("go", "env", "GOVERSION")
++		versionData, err = cmd.Output()
++		if err != nil {
++			return "", fmt.Errorf("failed to get Go toolchain version in GOROOT (%s)", goroot)
++		}
```
I don't have confidence in whether these patches are reasonable or not.
Is this patch only provided for Nix?
Don't we need to send it upstream and wait for it?

phanirithvij replied:
>  Is this patch only provided for Nix?

For nix only. I've tried it with archlinux and it succeeds.

```console
# arch linux
$ cat $(go env GOROOT)/VERSION
go1.23.3
time 2024-11-06T18:46:45Z

# go from nix
$ cat $(go env GOROOT)/VERSION
cat: /nix/store/bavnchxi7v6xs077jxv7fl5rrqc3y87w-go-1.23.3/share/go/VERSION: No such file or directory

$ go env GOVERSION
go1.23.3
```

The go package has no `<goroot>/VERSION` file in the build. Maybe it was removed for reproducibility?

But `go env GOVERSION` reports the same thing

kachick replied:
Sure, so I understood that this patch should be required for nixpkgs.

https://github.com/NixOS/nixpkgs/blob/6357f592127ca47fc21be5e8a11ec3e7b5572637/pkgs/README.md?plain=1#L510

>only do this for patches that are unique to `nixpkgs`

:heavy_check_mark:

kachick replied:
1. go introduced distpack https://github.com/golang/go/commit/1635205a72d26721af54f01ccbab8e0b51ded3a9, [it ensures GOROOT/VERSION](https://github.com/golang/go/blob/b68f8ca89a2fb6b33d1e78fadc33d1c35693f6b8/src/cmd/distpack/test.go#L43)
2. nixpkgs introduced it https://github.com/NixOS/nixpkgs/pull/254497
3. nixpkgs reverted it in https://github.com/NixOS/nixpkgs/pull/256588
4. nixpkgs have revised PR for that https://github.com/NixOS/nixpkgs/pull/256505 with draft state

phanirithvij replied:
Added a comment for the reason behind this patch.

kachick commented:
Do these 5 packages have relationships with each other?
I don't know the details of the nixpkgs convention. However, if there are different packages in one PR, I want to know the reason. :pray:

phanirithvij commented:
>  I don't know the details of the nixpkgs convention. However, if there are different packages in one PR, I want to know the reason. 🙏

I didn't want to make 5 prs when these all could be reviewed together, since the commits are all are self-contained. I've read the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) but didn't find anything which restricts each package update to a pr. But you are correct, if there was a relation between these it would be more reasonable.

I still feel that it would be inefficient and take forever with multiple prs, but I will do it separately from now on.

kachick commented:
Could you add `passthru.tests`?

https://github.com/NixOS/nixpkgs/blob/9e708b3c14db4ed50d3060406b2bc3c4eddd2fc0/doc/build-helpers/testers.chapter.md?plain=1#L168-L214

kachick commented on pkgs/by-name/pk/pkgsite/package.nix:
```diff
@@ -0,0 +1,35 @@
+  unstableGitUpdater,
+}:
+
+buildGo123Module rec {
```
Did you intentionally choose buildGo123Module for this package? Other Go packages in this PR are using buildGoModule.

phanirithvij replied:
Yes, it was a while ago since I wrote this derivation, the go version was 1.22 then in nixpkgs and pkgsite required go1.23+ in its go.mod. I will change it now.

phanirithvij commented:
I've updated passthru.updateScript to use `nix-update` instead of unstableGitUpdater, as I can run it my self from the command line.

> Could you add passthru.tests?

- Added version checks for gocovsh, golds.
- gama-tui has no cli, so we can't check for version.
- pkgsite has no version command in the cli.
- tcount has no tags (but internally it is version `0.1.0`), `nix-update` will overwrite the version to be `0-unstable-*` instead of `0.1.0-unstable-*`. Thus it is not possible to do version check for this without hardcoding the version inside the derviation or writing a custom update script.

kachick reviewed (approved):
I didn't experience these packages ever. However, the current difference in this PR looks good to me. :+1: 

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358316`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 29 packages built:</summary>
  <ul>
    <li>dbt (python312Packages.dbt-core)</li>
    <li>dbt.dist (python312Packages.dbt-core.dist)</li>
    <li>gama-tui</li>
    <li>gocovsh</li>
    <li>golds</li>
    <li>pkgsite</li>
    <li>python311Packages.dbt-bigquery</li>
    <li>python311Packages.dbt-bigquery.dist</li>
    <li>python311Packages.dbt-core</li>
    <li>python311Packages.dbt-core.dist</li>
    <li>python311Packages.dbt-postgres</li>
    <li>python311Packages.dbt-postgres.dist</li>
    <li>python311Packages.dbt-redshift</li>
    <li>python311Packages.dbt-redshift.dist</li>
    <li>python311Packages.dbt-semantic-interfaces</li>
    <li>python311Packages.dbt-semantic-interfaces.dist</li>
    <li>python311Packages.dbt-snowflake</li>
    <li>python311Packages.dbt-snowflake.dist</li>
    <li>python312Packages.dbt-bigquery</li>
    <li>python312Packages.dbt-bigquery.dist</li>
    <li>python312Packages.dbt-postgres</li>
    <li>python312Packages.dbt-postgres.dist</li>
    <li>python312Packages.dbt-redshift</li>
    <li>python312Packages.dbt-redshift.dist</li>
    <li>python312Packages.dbt-semantic-interfaces</li>
    <li>python312Packages.dbt-semantic-interfaces.dist</li>
    <li>python312Packages.dbt-snowflake</li>
    <li>python312Packages.dbt-snowflake.dist</li>
    <li>tcount</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>gama-tui</li>
    <li>gocovsh</li>
    <li>golds</li>
    <li>pkgsite</li>
    <li>tcount</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>gama-tui</li>
    <li>gocovsh</li>
    <li>golds</li>
    <li>pkgsite</li>
    <li>tcount</li>
  </ul>
</details>

phanirithvij commented:
```console
$ nom build -f default.nix gama-tui tcount gocovsh golds pkgsite --print-out-paths --no-link
Finished at 09:21:12 after 6s
/nix/store/xmx7bg5rlpfgd24bn9z9g2052plhyh3p-gama-tui-1.1.4-unstable-2024-09-05
/nix/store/wma3fhfnynykr5ynwhq2h5jcpkh275bd-tcount-0-unstable-2023-04-20
/nix/store/wr7xwn521xqc99shc7mjgzshjpjs3818-gocovsh-0.6.1
/nix/store/v6l3qa58lmam0nvng32xp7a1rnn5382f-golds-0.7.1
/nix/store/20br3y2xixplzb0n3mfny6w92hk3lwrs-pkgsite-0-unstable-2024-11-28
```

phanirithvij commented:
@ofborg build gama-tui gocovsh golds pkgsite tcount

SuperSandro2000 commented on pkgs/by-name/ga/gama-tui/package.nix:
```diff
@@ -0,0 +1,40 @@
+  src = fetchFromGitHub {
+    owner = "termkit";
+    repo = "gama";
+    rev = "5063d35e48f3118fa793965fcbb931ef3b330f02";
```
Why is the release version not enough?

phanirithvij replied:
I honestly don't remember, I was using unstable version for a while locally.

changed to tagged release

SuperSandro2000 commented on pkgs/by-name/go/gocovsh/package.nix:
```diff
@@ -0,0 +1,43 @@
+
+  vendorHash = "sha256-Fb7BIWojOSUIlBdjIt57CSvF1a+x33sB45Z0a86JMUg=";
+
+  # check pdfcpu drv if really need these
+  # or write an update script
```
```suggestion
```

SuperSandro2000 commented on pkgs/by-name/go/golds/package.nix:
```diff
@@ -0,0 +1,40 @@
+    hash = "sha256-6YkyKJtSAFFYidMlZXSjNpjyIIaTlibg/QMMin/NbU0=";
+  };
+
+  # https://github.com/NixOS/nixpkgs/pull/358316#discussion_r1855322027
```
```suggestion
  # nixpkgs is not using the go distpack archive and missing a VERSION file in the source but we can use go env to get the same information 
  # https://github.com/NixOS/nixpkgs/pull/358316#discussion_r1855322027
```

SuperSandro2000 commented on pkgs/by-name/go/golds/package.nix:
```diff
@@ -0,0 +1,40 @@
+  };
+
+  meta = {
+    description = "An experimental Go local docs server/generator and code reader implemented with some fresh ideas";
```
```suggestion
    description = "Experimental Go local docs server/generator and code reader implemented with some fresh ideas";
```

SuperSandro2000 commented on pkgs/by-name/tc/tcount/package.nix:
```diff
@@ -0,0 +1,33 @@
+    description = "Count your code by tokens and patterns in the syntax tree. A tokei/scc/cloc alternative";
+    homepage = "https://github.com/rrethy/tcount";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```
```suggestion
```
that is the default already

SuperSandro2000 commented on pkgs/by-name/pk/pkgsite/package.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Official tool to extract and generate documentation for Go projects like pkg.go.dev";
+    homepage = "https://github.com/golang/pkgsite";
+    license = lib.licenses.bsd3;
+    platforms = lib.platforms.unix;
```
buildGoModule defaults to this.

phanirithvij replied:
removed the `platforms` meta attribute.

phanirithvij commented:
@ofborg build gama-tui gocovsh golds pkgsite tcount
